docs/RIGHTS_DECLARATION.md

```diff
@@ -0,0 +1,23 @@
+Hypr Development Code Rights Declaration
+
+Last revision: 01 III 2024
+
+Copyright 2024, Hypr Development, vaxerski.
+
+/\ - /\ - /\ - /\
+
+TL;DR: I made this and I am okay with you guys relicensing this later as long as it's still open source.
+
+By making a contribution to this project, I certify that:
+
+ Ⅰ) This contribution is made entirely by myself, or in a way that
+ gives me full rights to the contribution.
+
+ Ⅱ) I realize and understand this contribution will be permanently recorded
+ under this repository alongside the information attached to it (Such as my Github username)
+
+ Ⅲ) I give full rights to Hypr Development to redistribute this contribution as a part of the parent project in both source and binary form.
+
+ Ⅳ) I give full rights to Hypr Development to relicense the parent project to which I am contributing at a later date to any license as long as that license does not hinder the ability of one to contribute to, or view the source code of the project.
```

Comment:
```suggestion
 Ⅳ) I give full rights to Hypr Development to relicense the parent project to which I am contributing at a later date to any license as long as that license does not hinder the ability of one to contribute to, redistribute or view the source code of the project.
```

Reply:
See I don't want to add that because imagine a scenario one day where a corporation starts using any part of my projects, and I want a cut from them. I have no leverage, as this disallows me from adding a clause prohibiting commercial use / use by commercial entities.

Reply:
In that case you will have to abandon the OSS definition i'm afraid.
For that usecase I would simply add non-commercial in front of it and be done with that.

I am in doubt if that holds up legally though.

Reply:
OSS definition? Where do I define as that? You mean line 9?

Also, I believe it's more like dropping the F rather than the OSS xD

How about `license does not hinder the ability of any non-commercial entity to contribute to, redistribute or view the source code of the project.`

Reply:
I guess that would do it.

About the OSS definition, I based that on the open source initiative, I would have pasted the section about this, but somehow they managed to let their certificate expire...... they pulled a manjaro

Reply:
In all honesty - based on a lot of the open source initiative points I've seen - it's a damn pipe dream and completely unrealistic for the modern world. You will _not_ get rid of the tech giants fucking people over in any way other than by law.

Reply:
committed, thanks!